cluster.go

```diff
@@ -241,3 +252,73 @@ var (
 	ErrNoConnectionsStarted = errors.New("no connections were made when creating the session")
 	ErrHostQueryFailed      = errors.New("unable to populate Hosts")
 )
+
+func NewCloudCluster(bundlePath string) (*ClusterConfig, error) {
+	connConf := &cloud.ConnectionConfig{}
+
+	bundleFile, err := os.Open(bundlePath)
+	if err != nil {
+		return nil, fmt.Errorf("can't open bundle path: %w", err)
+	}
+	defer bundleFile.Close()
+
+	if err := yaml.NewDecoder(bundleFile).Decode(connConf); err != nil {
+		return nil, fmt.Errorf("can't decode bundle file at %q: %w", bundlePath, err)
+	}
+
+	if _, ok := connConf.Contexts[connConf.CurrentContext]; !ok {
+		return nil, fmt.Errorf("current context points to unknown context")
+	}
+
+	confContext := connConf.Contexts[connConf.CurrentContext]
+
+	if _, ok := connConf.AuthInfos[confContext.AuthInfoName]; !ok {
+		return nil, fmt.Errorf("context %q auth info points to unknown authinfo", connConf.CurrentContext)
+	}
+
+	if _, ok := connConf.Datacenters[confContext.DatacenterName]; !ok {
+		return nil, fmt.Errorf("context %q datacenter points to unknown datacenter", connConf.CurrentContext)
+	}
+
+	authInfo := connConf.AuthInfos[confContext.AuthInfoName]
+
+	caPool, err := connConf.GetRootCAPool()
+	if err != nil {
+		return nil, fmt.Errorf("can't create root CA pool: %w", err)
+	}
+
+	cc := NewCluster(connConf.GetInitialContactPoints()...)
+	cc.Port = 443
+
+	// SslOpts are used only by establishing connection to initial contact points.
+	// Skip verifying TLS if any of DC requires it.
+	insecureSkipVerify := false
+	for _, dc := range connConf.Datacenters {
+		insecureSkipVerify = insecureSkipVerify || dc.InsecureSkipTLSVerify
```

Comment:
```suggestion
		insecureSkipVerify = dc.InsecureSkipTLSVerify
```

Reply:
It's not what I want here: 
```
// Skip verifying TLS if any of DC requires it
```

Reply:
i could break earlier though, ok i will change it

Reply:
done